tutorials/hoon/graph.md

```diff
@@ -0,0 +1,153 @@
+
+`marked-graph` is a pair of `graph` and `(unit mark)`. The semantics of a graph are defined by the mark. This is just for validating the structure of the graph. 
+
+`graph` is a `mop`, a mold builder for order map, of an atom and a `node`.
+
+`node` is a pair of `post` and `children`.
+
+`post` is defined in `sur/post`. It contains:
+
+`author=ship`
+`index`
+`time-sent=time`
+`contents=(list content)`
+`hash=(unit hash)`
+`signatures`
+
+Most of these elements are self explanatory.
+
+`contents` is a `list` of `content` where `content` is a tagged union of text, url, code, or reference. This is where the contents of a publish post or an entry from links would be stored.
+
+`hash` is a `sham` hash of `validated-portion` which includes the parent hash, author, time sent, and contents.
+
+`signatures` is a set of `signature` which is used to cryptographically sign graph-store messages.
+
+When interacting with `graph-store` you will need to create an `update`
+
+`update` is a tagged union that currently only has one option
+
+`[%0 p=time q=update-0]`
+
+The reason this is a union is to serve as future proofing for version changes.
+
+An `update-0` can either be logged in the case of adding and removing nodes or signatures or non logged as in the case of adding or removing a graph. This logging refers to if this action adds an entry to the update log that is used to synchronize state between ships.
+
+Below is an excerpt or `sur/graph-store.hoon` at the time of writing showing the definition of `logged-update-0` and `update-0` which should now be readable.
+
+```
++$  logged-update-0
+  $%  [%add-nodes =resource nodes=(map index node)]
+      [%remove-nodes =resource indices=(set index)]
+      [%add-signatures =uid =signatures]
+      [%remove-signatures =uid =signatures]
+  ==
+::
++$  update-0
+  $%  logged-update-0
+      [%add-graph =resource =graph mark=(unit mark)]
+      [%remove-graph =resource]
+    ::
+      [%add-tag =term =resource]
+      [%remove-tag =term =resource]
+    ::
+      [%archive-graph =resource]
+      [%unarchive-graph =resource]
+      [%run-updates =resource =update-log]
+    ::
+    ::  NOTE: cannot be sent as pokes
+    ::
+      [%keys =resources]
+      [%tags tags=(set term)]
+      [%tag-queries =tag-queries]
+  ==
+```
+
+
+## Pokes
+
+Graph store can be poked with a `graph-update` which can:
+ - add and remove a graph
+ 
+ `[%add-graph =resource =graph mark=(unit mark)]`
+ 
+ `[%remove-graph =resource]`
+ 
+ - add and remove a node to a graph
+ 
+ `[%add-nodes =resource nodes=(map index node)]`
+
+ `[%remove-nodes =resource indices=(set index)]`
+ 
+ - add and remove signatures
+ 
+ `[%add-signatures =uid =signatures]`
+ 
+ `[%remove-signatures =uid =signatures]`
+ 
+ - add and remove tags
+ 
+ `[%add-tag =term =resource]`
+ 
+ `[%remove-tag =term =resource]`
+ 
+ - archive and unarchive a graph
+ 
+ `[%archive-graph =resource]`
+ 
+ `[%unarchive-graph =resource]`
+
+## Scrys
```

Comment:
The plural of scry is scries, but since scry is a special word for us I'm not sure if that applies here

Reply:
Eh. let's go with scries for now.